FWCore/Framework/src/Schedule.cc

```diff
@@ -785,6 +785,23 @@ namespace edm {
 
     if (wantSummary_ == false)
       return;
+
+    //Function to loop over items in a container and periodically
+    // flush to the message logger.
+    auto logForEach = [](auto const& iContainer, auto iMessage) {
+      auto logger = LogFwkVerbatim("FwkSummary");
+      int count = 0;
+      for (auto const& element : iContainer) {
+        iMessage(logger, element);
+        if (++count == 10) {
```

Comment:
Why not appending all the elements of `iContainer` to the `logger`? To avoid constructing a huge string?

Reply:
Yes. I was concerned we could create a several MB sized string for the HLT cases. Given we have seen memory spikes at the end of the job, I didn't want to accidentally add to that.